tests/spicy/types/unit/block.spicy

```diff
@@ -0,0 +1,54 @@
+# @TEST-EXEC: ${SPICYC} %INPUT -j -o %INPUT.hlto
+# @TEST-EXEC: ${SCRIPTS}/printf '\x00\x10\x20\xff' | spicy-driver %INPUT.hlto >>output
+# @TEST-EXEC: ${SCRIPTS}/printf '\x01a1a2\x10\x20\xff' | spicy-driver %INPUT.hlto >>output
+# @TEST-EXEC: ${SCRIPTS}/printf '\x02\x10b2b3\x10\x20\xff' | spicy-driver %INPUT.hlto >>output
+# @TEST-EXEC: ${SCRIPTS}/printf '\x02\x20\x0a\x10\x20\xff' | spicy-driver %INPUT.hlto >>output
+# @TEST-EXEC: ${SCRIPTS}/printf '\x03d1\xff' | spicy-driver %INPUT.hlto >>output
+# @TEST-EXEC: btest-diff output
+#
+# @TEST-DOC: Checks unit if-blocks.
+
+module Mini;
+
+public type X = unit {
```

Comment:
Interesting tidbit: the parser currently permits putting unit vars (which are also just items) in these new blocks which might allow to more cleanly scope code (only for folding, items in syvblock are still visible in whole unit). Should we add a test for that to make sure to not break it?

If one introduces the same var in multiple blocks the error message is currently not very helpful since we report the full generated struct range (and also talk about attributes). Incorrect code like

```ruby
    if (True) {
        var myvar: uint8;
    } else {
        var myvar: uint8;
    };
```
produces

```
block.spicy:13:1-63:2: duplicate attribute in struct type
```

Reply:
I think I'd rather reject because initialization semantics become muddy. But now new, I suppose this is the same already with switches.

Reply:
Adding a validator check.